app/plugins.go

```diff
@@ -54,6 +57,67 @@ func (api *PluginAPI) CreatePost(post *model.Post) (*model.Post, *model.AppError
 	return CreatePostMissingChannel(post, true)
 }
 
+func (api *PluginAPI) GetLdapUserAttributes(userId string, attributes []string) (map[string]string, *model.AppError) {
+	ldapInterface := einterfaces.GetLdapInterface()
+	if ldapInterface == nil {
+		return nil, model.NewAppError("GetLdapUserAttributes", "ent.ldap.disabled.app_error", nil, "", http.StatusNotImplemented)
+	}
+
+	user, err := GetUser(userId)
+	if err != nil {
+		return nil, err
+	}
+
+	return ldapInterface.GetUserAttributes(*user.AuthData, attributes)
+}
+
+func (api *PluginAPI) GetSessionFromRequest(r *http.Request) (*model.Session, *model.AppError) {
```

Comment:
I'm a little surprised that we don't already have a utility function for grabbing a user's session out of a request object. It seems like the sort of thing that some other endpoint would have required before now.

Reply:
Yeah, I considered using this in the api and api4 packages but it's way too late in the release cycle for a change to that code